pants

```diff
@@ -41,17 +41,25 @@ source "${HERE}/build-support/pants_venv"
 source "${HERE}/build-support/bin/rust/bootstrap_code.sh"
 
 function exec_pants_bare() {
-  PANTS_NATIVE_EXE="${HERE}/.pants"
   PANTS_PY_EXE="${HERE}/src/python/pants/bin/pants_loader.py"
   PANTS_SRCPATH="${HERE}/src/python"
 
   # Redirect activation and native bootstrap to ensure that they don't interfere with stdout.
   activate_pants_venv 1>&2
   bootstrap_native_code 1>&2
 
-  if [ -n "${USE_NATIVE_PANTS}" ]; then
+  if [ -n "${PANTS_DEBUG}" ]; then
+    if [[ "$*" != *"--no-pantsd"* ]]; then
+      echo "Error! Must pass '--no-pantsd' when using PANTS_DEBUG"
+      exit 1
+    fi
+    DEBUG_ARGS="-m debugpy --listen 127.0.0.1:5678 --wait-for-client"
+    echo "Will launch debugpy server at '127.0.0.1:5678' waiting for client connection."
+  fi
+
+  if [ -z "${PANTS_NO_NATIVE_CLIENT}" ]; then
     set +e
-    "${PANTS_NATIVE_EXE}" "$@"
+    "${NATIVE_CLIENT_BINARY}" "$@"
```

Comment:
Did you intend to get some mileage on _PANTS_SERVER_EXE here? Seems good to do.

Reply:
I didn't (yet): it's mentioned in the description as something that would be worth doing. But I think that doing it requires significant refactoring of this script to split it into `pants` vs `pants_server` scripts.